Sources/AWSLambdaEvents/Codable Helpers/APIGatewayV2+Encode.swift

```diff
@@ -0,0 +1,41 @@
+#if canImport(FoundationEssentials)
+import FoundationEssentials
+#else
+import Foundation
+#endif
+
+import HTTPTypes
+
+public extension APIGatewayV2Response {
+    /// Encodes a given encodable object into a `APIGatewayV2Response` object.
+    ///
+    /// - Parameters:
+    ///   - encodable: The object to encode.
+    ///   - status: The status code to use. Defaults to `ok`.
+    ///   - encoder: The encoder to use. Defaults to a new `JSONEncoder`.
+    ///   - onError: A closure to handle errors, and transform them into a `APIGatewayV2Response`. Defaults
+    /// to converting the error into a 500 (Internal Server Error) response with the error message as the body.
+    static func encoding<T>(
+        _ encodable: T,
+        status: HTTPResponse.Status = .ok,
+        using encoder: JSONEncoder = JSONEncoder(),
+        onError: ((Error) -> Self)? = nil
```

Comment:
just curious : why not make `encoding<T>` throwing ?

Reply:
OK, I understand and I think it's a good idea.  What about providing two versions of this function? One trowing (without error callback) and one with the error callback for ease of use as you designed. The later would just call the throwing version + add default error handling.

Reply:
Sounds good!